e2e/interchain_accounts_test.go

```diff
@@ -0,0 +1,220 @@
+package e2e
+
+import (
+	"context"
+	"testing"
+
+	ibctest "github.com/strangelove-ventures/ibctest"
+	"github.com/strangelove-ventures/ibctest/chain/cosmos"
+	"github.com/strangelove-ventures/ibctest/ibc"
+	"github.com/strangelove-ventures/ibctest/test"
+	"github.com/stretchr/testify/suite"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	intertxtypes "github.com/cosmos/interchain-accounts/x/inter-tx/types"
+
+	"github.com/cosmos/ibc-go/e2e/testconfig"
+	"github.com/cosmos/ibc-go/e2e/testsuite"
+	"github.com/cosmos/ibc-go/e2e/testvalues"
+)
+
+func TestInterchainAccountsTestSuite(t *testing.T) {
+	// NOTE: this is a temporary mechanism to enable this test to run alongside the simd tests.
+	// This will be removed in a follow up PR and properly parameterized in a github workflow.
+	testconfig.SetChainBinaryVersions(
+		"ghcr.io/cosmos/ibc-go-icad", "master", "icad", "ghcr.io/cosmos/ibc-go-icad", "master",
+	)
+	suite.Run(t, new(InterchainAccountsTestSuite))
+}
+
+type InterchainAccountsTestSuite struct {
+	testsuite.E2ETestSuite
+}
+
+// RegisterICA will attempt to register an interchain account on the counterparty chain.
+func (s *InterchainAccountsTestSuite) RegisterICA(ctx context.Context, chain *cosmos.CosmosChain, user *ibctest.User, fromAddress, connectionID string) error {
+	version := "" // allow app to handle the version as appropriate.
+	msg := intertxtypes.NewMsgRegisterAccount(fromAddress, connectionID, version)
+	txResp, err := s.BroadcastMessages(ctx, chain, user, msg)
+	s.AssertValidTxResponse(txResp)
+	return err
+}
+
+func (s *InterchainAccountsTestSuite) TestMsgSubmitTx_SuccessfulTransfer() {
+	t := s.T()
+	ctx := context.TODO()
+
+	// setup relayers and connection-0 between two chains
+	// channel-0 is a transfer channel but it will not be used in this test case
+	relayer, _ := s.SetupChainsRelayerAndChannel(ctx)
+	chainA, chainB := s.GetChains()
+
+	connectionId := "connection-0"
+
+	// setup 2 accounts: controller account on chain A, a second chain B account.
+	// host account will be created when the ICA is registered
+	controllerAccount := s.CreateUserOnChainA(ctx, testvalues.StartingTokenAmount)
```

Comment:
wdyt about renaming `ControllerAccount` to `OwnerAccount` and the `hostAccount` as `InterchainAccount`?

cc: @damiannolan

Reply:
i personally think that this change would be more confusing -- but that's just my opinion

Reply:
I personally like host/controller terminology